FeedStoreChallenge/RealmFeedStore.swift

```diff
@@ -0,0 +1,134 @@
+//
+//  RealmFeedStore.swift
+//  FeedStoreChallenge
+//
+//  Created by Salah Amassi on 27/02/2021.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import Foundation
+import RealmSwift
+
+public class RealmFeedStore: FeedStore {
+	
+	let storeURL: URL?
+	
+	public init(storeURL: URL?) {
+		self.storeURL = storeURL
+	}
+	
+	public func retrieve(completion: @escaping RetrievalCompletion) {
+		do {
+			let realm = try getRealm()
+			if let cache = realm.objects(Cache.self).last {
+				completion(.found(feed: cache.feed, timestamp: cache.timestamp))
+			} else {
+				completion(.empty)
+			}
+		} catch Realm.Error.fileNotFound, Realm.Error.filePermissionDenied {
+			completion(.empty)
+		} catch {
+			completion(.failure(error))
+		}
```

Comment:
If there's *any* error, the store should complete with `completion(.failure(error))`.

```suggestion
		} catch {
			completion(.failure(error))
		}
```